infoblox_client/utils.py

```diff
@@ -32,19 +31,27 @@ def is_valid_ip(ip):
     return True
 
 
+def get_macstr_from_mac(mac):
+    return ''.join(c for c in mac if c not in ':')
+
+
 def generate_duid(mac):
     """DUID is consisted of 10 hex numbers.
 
-    0x00 + 3 random hex + mac with 6 hex
+    0x00 + 3 hex from last 3 character of mac_str + mac with 6 hex
```

Comment:
Thanks for the explanation!

You can decide how to handle this. These would be my considerations:
- Meet the need of the code, i.e. guaranteed uniqueness to be used for search
- Meet relevant standards
- Meet NIOS requirement
- The simpler the better.

Note also that, except maybe for troubleshooting, whether the duid "looks like" unique is not important.

Thanks again!